### PR TITLE
Improving UX & Code

### PR DESCRIPTION
**Pull request by two-one-five**
I want to continue to clean out the code but also improve UX, catch any edge cases if there might be any to a new user.

### CODE REVIEW

File: src/background.js [hunk 1]
```diff
@@ -400,17 +404,19 @@ ipcMain.on('save-state', (event, arg) => {
 })
 
 ipcMain.on('load-state', (event, arg) => {
-	getSetting('vircadia_launcher.state', storagePath.default).then(function(results){
-		if(results) {
+	getSetting('vircadia_launcher.state', storagePath.default).then(function(results) {
+		if (results) {
```
Comment (ctrlaltdavid) [inline src/background.js]: Tabs, here and elsewhere.

File: src/components/Dialogs/FirstTimeUser.vue [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                primary-title
+                dark
+            >
+                <v-icon color="green" class="mr-2">mdi-emoticon-excited-outline</v-icon>
+                Are you new?
+            </v-card-title>
+    
+            <v-card-text>
+                Hello there! If you're new to Vircadia, you'll probably want to get the Interface.
+                The Interface is what you will use to navigate and enjoy the virtual worlds in desktop or VR!
```
Comment (ctrlaltdavid) [inline src/components/Dialogs/FirstTimeUser.vue]:
```suggestion
                The Interface is what you use to navigate and enjoy the virtual worlds in desktop or VR!
```

File: vue.config.js [hunk 3]
```diff
@@ -25,8 +25,9 @@ module.exports = {
 				"win": {
 					"target": "nsis",
 					"icon": "src/assets/logo_256_256.ico",
-					"publisherName": "https://vircadia.com/",
+					"publisherName": "Vircadia",
```
Comment (ctrlaltdavid) [inline vue.config.js]: Tabs.

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

File: src/background.js [hunk 4]
```diff
@@ -375,84 +376,84 @@ async function getDownloadURL() {
...
-	})
+    var returnValue;
+
+    let storagePromise = new Promise((res, rej) => {
+        storage.get(setting, {dataPath: storageDataPath}, function(error, data) {
+            if (error) {
+                returnValue = false;
+                rej("Error: " + error);
+                throw error;
+            } else if (Object.entries(data).length == 0) {
```
Comment (ctrlaltdavid) [inline src/background.js]:
```suggestion
            } else if (Object.entries(data).length === 0) {
```

File: src/background.js [hunk 5]
```diff
@@ -140,7 +141,7 @@ var storagePath = {
 	currentLibrary: null,
 };
 
-var installFolderName = "\\Vircadia_Interface_Latest_SILENT";
+var installFolderName = "\\Vircadia_Interface_Latest_SILENT/";
```
Comment (ctrlaltdavid) [inline src/background.js]:
```suggestion
var installFolderName = "\\Vircadia_Interface_Latest_SILENT\\";
```
Or both as "/"?

File: src/background.js [hunk 6]
```diff
@@ -375,84 +376,84 @@ async function getDownloadURL() {
...
 })
 
 ipcMain.on('set-metaverse-server', (event, arg) => {
-	if (arg != "") {
-		process.env.HIFI_METAVERSE_URL = arg;
-	} else {
-		delete process.env.HIFI_METAVERSE_URL;
-	}
-	console.info("Current Metaverse Server:", process.env.HIFI_METAVERSE_URL)
+    if (arg != "") {
```
Comment (ctrlaltdavid) [inline src/background.js]:
```suggestion
    if (arg !== "") {
```

File: src/background.js [hunk 7]
```diff
@@ -722,10 +729,24 @@ ipcMain.on('download-vircadia', async (event, arg) => {
...
-                if (err) console.log("No previous installation to delete.");
-                console.info(installerName, 'was deleted prior to downloading.');
-            });
+            var previousInstaller = libraryPath + "/" + installerNamePost;
+            
+            if (fs.existsSync(previousInstaller)) {
+                var md5current = hasha.fromFileSync(previousInstaller, {algorithm: 'md5'});
+                md5current = md5current.toUpperCase();
+                
+                if (md5current == vircadiaMetaJSON.latest.md5) {
```
Comment (ctrlaltdavid) [inline src/background.js]:
```suggestion
                if (md5current === vircadiaMetaJSON.latest.md5) {
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

File: src/App.vue [hunk 8]
```diff
@@ -287,35 +295,54 @@ ipcRenderer.on('download-installer-failed', (event) => {
...
+	if (arg.results.noSteamVR) {
 		vue_this.$store.commit('mutate', {
 			property: 'noSteamVR', 
 			with: arg.results.noSteamVR
 		});
 		vue_this.noSteamVR = arg.results.noSteamVR;
 	}
-	if(arg.results.allowMultipleInstances) {
+    
+    if (arg.results.sentryEnabled != null) {
```
Comment (ctrlaltdavid) [inline src/App.vue]:
```suggestion
    if (arg.results.sentryEnabled !== null) {
```

File: src/App.vue [hunk 9]
```diff
@@ -287,35 +295,54 @@ ipcRenderer.on('download-installer-failed', (event) => {
...
+		vue_this.$store.commit('mutate', {
+			property: 'sentryEnabled', 
+			with: arg.results.sentryEnabled
+		});
+	}
+    
+    if (arg.results.darkMode != null) {
```
Comment (ctrlaltdavid) [inline src/App.vue]:
```suggestion
    if (arg.results.darkMode !== null) {
```

File: src/background.js [hunk 10]
```diff
@@ -650,7 +656,11 @@ async function silentInstall(useOldInstaller) {
                     if (err.code === "EACCES") {
                         errorMessage = "Please run the launcher as an administrator to continue.";
                     } else {
-                        errorMessage = "An error has occurred with code: " + err.code + " Full Error: " + err;
+                        if (err.code == 2) {
```
Comment (ctrlaltdavid) [inline src/background.js]:
```suggestion
                        if (err.code === 2) {
```

File: src/components/Settings.vue [hunk 11]
```diff
@@ -354,6 +387,10 @@ export default {
             this.selectedInterface = this.$store.state.selectedInterface.name;
         }
         
+        if (this.$store.state.sentryEnabled != null) {
```
Comment (ctrlaltdavid) [inline src/components/Settings.vue]:
```suggestion
        if (this.$store.state.sentryEnabled !== null) {
```

File: src/plugins/vuetify.js [hunk 12]
```diff
@@ -14,4 +14,7 @@ import Vuetify from 'vuetify/lib';
 Vue.use(Vuetify);
 
 export default new Vuetify({
+    theme: {
+        dark: true,
```
Comment (ctrlaltdavid) [inline src/plugins/vuetify.js:20]:
```suggestion
        dark: true
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Comment by two-one-five** [comment]
I think I want to improve the layout that errors have in the window to at least be easier to follow and read. Also, the color scheme needs a bit of work in terms of buttons and title backgrounds as is highlighted when you switch to the dark theme.

**Comment by two-one-five** [comment]
This PR may or may not have expanded outside of the scope of its original intent. Oh well. ;)

File: public/css/styles.css [hunk 13]
```diff
@@ -20,10 +20,20 @@ h2 {
   font-family: 'Raleway', sans-serif;
 }
 
-#titleURL {
-	color: white;
+.titleURL {
+	/* color: white; */
```
Comment (ctrlaltdavid) [inline public/css/styles.css]: Remove commented-out line?

File: public/css/styles.css [hunk 14]
```diff
@@ -46,22 +55,42 @@ h2 {
...
+    animation: gradientBG 15s ease infinite; */
+
+    background-image: url('../resources/mtnbg.jpg');
+    background-attachment: fixed;
+    background-position: center;
+    background-repeat: no-repeat;
+    background-size: cover;
+    max-width: none !important;
+    
+    height: 100vh;
```
Comment (ctrlaltdavid) [inline public/css/styles.css:65]: Is `100vh` valid CSS?

File: public/css/styles.css [hunk 15]
```diff
@@ -46,22 +55,42 @@ h2 {
 	z-index: 99;
 }
 
+#background {
+    /* background: linear-gradient(90deg, #00d2ff 0%, #3a47d5 100%);
+    background-size: 400% 400%;
+    animation: gradientBG 15s ease infinite; */
```
Comment (ctrlaltdavid) [inline public/css/styles.css]: Remove commented-out lines?

File: public/css/styles.css [hunk 16]
```diff
@@ -46,22 +55,42 @@ h2 {
...
+    padding-bottom: 0px !important;
+}
+
 #mainContent {	  
-	/* background: linear-gradient(90deg, #00d2ff 0%, #3a47d5 100%);
-	background-size: 400% 400%;
-	animation: gradientBG 15s ease infinite; */
+    box-sizing: border-box;
+    /* padding-top: 81px !important;
+    padding-bottom: 64px !important; */
```
Comment (ctrlaltdavid) [inline public/css/styles.css]: Remove commented-out lines?

File: public/css/styles.css [hunk 17]
```diff
@@ -90,9 +119,21 @@ pre {
 	margin-bottom: 6px;
 	padding: 10px;
 	font-size: 14px;
+    overflow: auto;
```
Comment (ctrlaltdavid) [inline public/css/styles.css:115]: Indentation. (Other lines have tabs; they should be spaces.)

File: public/index.html [hunk 18]
```diff
@@ -14,9 +14,10 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width,initial-scale=1.0">
     <link rel="icon" href="<%= BASE_URL %>/icon/logo_256_256.ico">
-    <title>Vircadia Launcher</title>
+    <title>Vircadia Launcher v0.0.1k</title>
```
Comment (ctrlaltdavid) [inline public/index.html]: Version number doesn't match that in package.json.

File: src/App.vue [hunk 19]
```diff
@@ -20,26 +25,51 @@
...
+            </v-btn-toggle>
         </v-bottom-navigation>
+        
+    <v-snackbar
+        v-model="interfaceBusyLaunching"
+        style="margin-bottom: 75px;"
+        color="#34323e"
+        :timeout="interfaceBusyLaunchingTimeout"
+    >
+        The Interface is starting.
```
Comment (ctrlaltdavid) [inline src/App.vue]:
```suggestion
        Interface is starting.
```
Maybe.

File: src/App.vue [hunk 20]
```diff
@@ -244,66 +259,115 @@ ipcRenderer.on('download-installer-progress', (event, arg) => {
...
-		vue_this.noSteamVR = arg.results.noSteamVR;
 	}
-	if(arg.results.allowMultipleInstances) {
+    
+    if (arg.results.noOculus) {
+		vue_this.$store.commit('mutate', {
+			property: 'noOculus', 
+			with: arg.results.noOculus
+		});
+	}
```
Comment (ctrlaltdavid) [inline src/App.vue]: Tabs.

File: src/App.vue [hunk 21]
```diff
@@ -244,66 +259,115 @@ ipcRenderer.on('download-installer-progress', (event, arg) => {
...
+            with: arg.results.customLaunchParameters
+        });
+    }
+    
+    if (arg.results.sentryEnabled !== null) {
+		vue_this.$store.commit('mutate', {
+			property: 'sentryEnabled', 
+			with: arg.results.sentryEnabled
+		});
+	}
```
Comment (ctrlaltdavid) [inline src/App.vue]: Tabs.

File: src/App.vue [hunk 22]
```diff
@@ -244,66 +259,115 @@ ipcRenderer.on('download-installer-progress', (event, arg) => {
...
+	if (arg.results.allowMultipleInstances) {
 		vue_this.$store.commit('mutate', {
 			property: 'allowMultipleInstances', 
 			with: arg.results.allowMultipleInstances
 		});
-        vue_this.allowMultipleInstances = arg.results.allowMultipleInstances;
 	}
-	if(arg.results.selectedInterface) {
+    
+    if (arg.results.autoRestartInterface) {
```
Comment (ctrlaltdavid) [inline src/App.vue:341]: Tabs below, also, throughout file.

File: src/App.vue [hunk 23]
```diff
@@ -478,13 +627,38 @@ export default {
             ipcRenderer.send('set-vircadia-location');
         },
         checkForUpdates: function() {
+            this.disableUpdateButton = true;
+            this.disableLaunchButton = true;
+            
             const { ipcRenderer } = require('electron');
             ipcRenderer.send('check-for-updates');            
+        },
+        resetDownloadButton: function() {
```
Comment (ctrlaltdavid) [inline src/App.vue]:
```suggestion
        resetDownloadButton: function () {
```
Should have a space between `function` and `(` in anonymous functions, here and throughout file.

File: src/background.js [hunk 24]
```diff
@@ -10,26 +10,32 @@
 
 'use strict'
 
+// import * as Sentry from '@sentry/electron'
+// import { init } from '@sentry/electron/dist/main';
+// init({dsn: 'https://def94db0cce14e2180e054407e551220@sentry.vircadia.dev/3'});
```
Comment (ctrlaltdavid) [inline src/background.js:15]: Remove commented-out code?

File: src/background.js [hunk 25]
```diff
@@ -10,26 +10,32 @@
...
+
 import { app, protocol, BrowserWindow, DownloadItem } from 'electron'
 import {
 	installVueDevtools,
 	createProtocol,
 } from 'vue-cli-plugin-electron-builder/lib'
 import path from 'path'
-const isDevelopment = process.env.NODE_ENV !== 'production'
+var forceDevelopment = false;
```
Comment (ctrlaltdavid) [inline src/background.js]:
```suggestion
const forceDevelopment = false;
```
const?

File: src/background.js [hunk 26]
```diff
@@ -136,13 +142,11 @@ var storagePath = {
 	currentLibrary: null,
 };
 
-var installFolderName = "\\Vircadia_Interface_Latest_SILENT";
+// var installFolderName = "\\Vircadia_Interface_Latest_SILENT\\";
```
Comment (ctrlaltdavid) [inline src/background.js]: Remove commented-out code?

File: src/background.js [hunk 27]
```diff
@@ -180,33 +184,33 @@ async function getDirectories (src) {
...
-				res_p();
-			} else {
-				rej_p("Failed to load directories.");
-			}
-		});
-		
-        // THIS CODE ONLY LOOKS FOR INTERFACE.EXE
-        
-		// glob(src + '/*/interface.exe', function(err, folders) {
+		// glob(src + '/*/launcher_settings/interface_package.json', function(err, folders) {
```
Comment (ctrlaltdavid) [inline src/background.js:185]: Remove commented-out code?

File: src/background.js [hunk 28]
```diff
@@ -180,33 +184,33 @@ async function getDirectories (src) {
...
 		// 	} else {
 		// 		rej_p("Failed to load directories.");
 		// 	}
 		// });
+		
+        // THIS CODE ONLY LOOKS FOR INTERFACE.EXE
+        
+		glob(src + '/*/interface.exe', function(err, folders) {
+			console.log(folders);
+			if(folders) {
```
Comment (ctrlaltdavid) [inline src/background.js]:
```suggestion
			if (folders) {
```

File: src/background.js [hunk 29]
```diff
@@ -180,33 +184,33 @@ async function getDirectories (src) {
...
 		// 		});
 		// 		res_p();
 		// 	} else {
 		// 		rej_p("Failed to load directories.");
 		// 	}
 		// });
+		
+        // THIS CODE ONLY LOOKS FOR INTERFACE.EXE
+        
+		glob(src + '/*/interface.exe', function(err, folders) {
```
Comment (ctrlaltdavid) [inline src/background.js]: Tabs. Here and elsewhere.

File: src/background.js [hunk 30]
```diff
@@ -271,18 +278,26 @@ function setLibraryDialog() {
...
-    let interfaceJSON = JSON.parse(rawdata);
-    
-    if (interfaceJSON) {
-        console.info("Interface Package JSON:", interfaceJSON);
-        return interfaceJSON;
-    } else {
-        return false;
-    }
-}
+// async function getCurrentInterfaceJSON() {
```
Comment (ctrlaltdavid) [inline src/background.js:279]: Remove commented-out code?

File: src/background.js [hunk 31]
```diff
@@ -325,7 +340,11 @@ async function getLatestMetaJSON() {
 
 async function checkForInterfaceUpdates() {
 	var vircadiaMeta = await getLatestMetaJSON();
-    var interfacePackage = await getCurrentInterfaceJSON();
+    // var interfacePackage = await getCurrentInterfaceJSON();
```
Comment (ctrlaltdavid) [inline src/background.js:341]: Remove commented-out code?

File: src/background.js [hunk 32]
```diff
@@ -364,89 +413,151 @@ async function getDownloadURL() {
...
+    console.info("Current Metaverse Server:", process.env.HIFI_METAVERSE_URL)
 })
 
-ipcMain.on('launch-interface', (event, arg) => {
-	var interface_exe = require('child_process').execFile;
-	// var executablePath = "E:\\Development\\High_Fidelity\\v0860-kasen-VS-release+freshstart\\build\\interface\\Packaged_Release\\Release\\interface.exe";
-	var executablePath = arg.exec;
-	var parameters = [];
+ipcMain.on('launch-interface', async (event, arg) => {
+    // var executablePath = "E:\\Development\\High_Fidelity\\v0860-kasen-VS-release+freshstart\\build\\interface\\Packaged_Release\\Release\\interface.exe";
```
Comment (ctrlaltdavid) [inline src/background.js]: Remove commented-out code?

File: src/background.js [hunk 33]
```diff
@@ -364,89 +413,151 @@ async function getDownloadURL() {
...
+                returnValue = false;
+                rej("Not found.")
+            } else {
+                returnValue = data;
+                res("Success!");
+            }
+        });
+    }).catch(err => {
+        console.info("Attempted to retrieve:", setting, "from:", storageDataPath, "but got:", err)
+    })
```
Comment (ctrlaltdavid) [inline src/background.js]:
```suggestion
    });
```

File: src/background.js [hunk 34]
```diff
@@ -364,89 +413,151 @@ async function getDownloadURL() {
...
 
-	// because async won't work otherwise. 
-	let result = await storagePromise; 
-	console.info("getSetting Return Value:",returnValue)
-	return returnValue;
+    // because async won't work otherwise. 
+    let result = await storagePromise; 
+    console.info("getSetting Return Value:",returnValue)
```
Comment (ctrlaltdavid) [inline src/background.js]:
```suggestion
    console.info("getSetting Return Value:", returnValue);
```

File: src/background.js [hunk 35]
```diff
@@ -364,89 +413,151 @@ async function getDownloadURL() {
...
-	} else {
-		delete process.env.HIFI_METAVERSE_URL;
-	}
-	console.info("Current Metaverse Server:", process.env.HIFI_METAVERSE_URL)
+    if (arg !== "") {
+        process.env.HIFI_METAVERSE_URL = arg;
+    } else {
+        delete process.env.HIFI_METAVERSE_URL;
+    }
+    console.info("Current Metaverse Server:", process.env.HIFI_METAVERSE_URL)
```
Comment (ctrlaltdavid) [inline src/background.js]:
```suggestion
    console.info("Current Metaverse Server:", process.env.HIFI_METAVERSE_URL);
```

File: src/background.js [hunk 36]
```diff
@@ -364,89 +413,151 @@ async function getDownloadURL() {
...
+    win.webContents.send('launching-interface');
+    
+    var interface_exe = require('child_process').execFile;
+    
+    interface_exe(executablePath, parameters, { windowsVerbatimArguments: true }, function(err, stdout, data) {
+        console.info("Interface.exe exited with code:", err);
+        if (autoRestartInterface == true && err && !err.killed) {
+            launchInterface(executablePath, parameters, autoRestartInterface);
+        }
+        // console.log(stdout.toString());
```
Comment (ctrlaltdavid) [inline src/background.js]: Remove commented-out code?

File: src/background.js [hunk 37]
```diff
@@ -518,20 +630,40 @@ ipcMain.handle('isInterfaceSelectionRequired', (event, arg) => {
...
-		var generatedList = await generateInterfaceList(results);
-		// console.info("Returning...", generatedList, "typeof", typeof generatedList, "results", results);
-		event.sender.send('interface-list-for-launch', generatedList);
-	});
+ipcMain.handle('get-interface-list-for-launch', async (event, arg) => {
+    var interface_exes = await getLibraryInterfaces();
+    var list = interface_exes.map(function(filename) {
+        // :)
+        var nv = versionPaths.fromPath(filename);
+        return { [nv.name]: { "location": filename.replace(/\binterface\.exe\b/i, '') } }
```
Comment (ctrlaltdavid) [inline src/background.js]:
```suggestion
        return { [nv.name]: { "location": filename.replace(/\binterface\.exe\b/i, '') } };
```

File: src/background.js [hunk 38]
```diff
@@ -518,20 +630,40 @@ ipcMain.handle('isInterfaceSelectionRequired', (event, arg) => {
...
-    getLibraryInterfaces().then(async function(results) {
-        var generatedList = await generateInterfaceList(results);
-        console.info("Returning...", generatedList, "typeof", typeof generatedList, "results", results);
-        event.sender.send('interface-list', generatedList);
+ipcMain.handle('populateInterfaceList', async (event, arg) => {
+    var interface_exes = await getLibraryInterfaces();
+    var list = interface_exes.map(function(filename) {
+        // :)
+        var nv = versionPaths.fromPath(filename);
+        return { [nv.name]: { "location": filename.replace(/\binterface\.exe\b/i, ''), "version": nv.version } }
```
Comment (ctrlaltdavid) [inline src/background.js]:
```suggestion
        return { [nv.name]: { "location": filename.replace(/\binterface\.exe\b/i, ''), "version": nv.version } };
```

File: src/background.js [hunk 39]
```diff
@@ -593,30 +725,34 @@ async function silentInstall() {
...
+                    console.info(executableLocation, 'was deleted after copying.');
+                });
+                
+                exeLocToInstall = executablePath + "/Vircadia_Setup_Latest_READY.exe";
+            }
         }
         
         win.webContents.send('silent-installer-running');
-
+
```
Comment (ctrlaltdavid) [inline src/background.js]:
```suggestion

```

File: src/background.js [hunk 40]
```diff
@@ -627,65 +763,92 @@ async function silentInstall() {
...
+                            errorMessage = "An error has occurred.";                
+                        }
+                    }
+                    
+                    win.webContents.send('silent-installer-failed', { "message": errorMessage, "code": err.code, "fullerr": err });
                 } else {
                     console.info("Installation complete.");
                     console.info("Running post-install.");
-                    postInstall();
+                    // postInstall();
```
Comment (ctrlaltdavid) [inline src/background.js:781]: Remove commented-out code.

File: src/background.js [hunk 41]
```diff
@@ -627,65 +763,92 @@ async function silentInstall() {
...
+                        errorMessage = "Please run the launcher as an administrator to continue.";
+                    } else {
+                        if (err.code === 2) {
+                            errorMessage = "An instance of Interface is running, please close it before proceeding.";
+                        } else {
+                            errorMessage = "An error has occurred.";                
+                        }
+                    }
+                    
+                    win.webContents.send('silent-installer-failed', { "message": errorMessage, "code": err.code, "fullerr": err });
```
Comment (ctrlaltdavid) [inline src/background.js]:
```suggestion
                    win.webContents.send('silent-installer-failed', { 
                        "message": errorMessage, 
                        "code": err.code, 
                        "fullerr": err 
                    });
```

File: src/components/Dialogs/CheckForUpdatesFailed.vue [hunk 42]
```diff
@@ -0,0 +1,74 @@
+<!--
+//  CheckForUpdatesFailed.vue
+//
+//  Created by Kalila L. on 22 Feb 2020.
```
Comment (ctrlaltdavid) [inline src/components/Dialogs/CheckForUpdatesFailed.vue]: Date correct?

File: src/components/Dialogs/CheckForUpdatesFailed.vue [hunk 43]
```diff
@@ -0,0 +1,74 @@
...
+//  Copyright 2020 Project Athena and contributors.
```
Comment (ctrlaltdavid) [inline src/components/Dialogs/CheckForUpdatesFailed.vue]:
```suggestion
//  Copyright 2020 Vircadia contributors.
```

File: src/components/Dialogs/FailedMetadata.vue [hunk 44]
```diff
@@ -0,0 +1,62 @@
+<!--
+//  FailedMetadata.vue
+//
+//  Created by KasenVR on 17 Apr 2020.
+//  Copyright 2020 Vircadia and contributors.
```
Comment (ctrlaltdavid) [inline src/components/Dialogs/FailedMetadata.vue]:
```suggestion
//  Copyright 2020 Vircadia contributors.
```

File: src/components/Dialogs/FirstTimeUser.vue [hunk 45]
```diff
@@ -0,0 +1,69 @@
+<!--
+//  FirstTimeUser.vue
+//
+//  Created by KasenVR on 13 April 2020.
+//  Copyright 2020 Vircadia and contributors.
```
Comment (ctrlaltdavid) [inline src/components/Dialogs/FirstTimeUser.vue]:
```suggestion
//  Copyright 2020 Vircadia contributors.
```

File: src/components/Dialogs/LaunchFailedInterfaceRunning.vue [hunk 46]
```diff
@@ -0,0 +1,58 @@
+<!--
+//  LaunchFailedInterfaceRunning.vue
+//
+//  Created by KasenVR on 15 May 2020.
+//  Copyright 2020 Vircadia and contributors.
```
Comment (ctrlaltdavid) [inline src/components/Dialogs/LaunchFailedInterfaceRunning.vue]:
```suggestion
//  Copyright 2020 Vircadia contributors.
```

File: src/components/Dialogs/LaunchOptions.vue [hunk 47]
```diff
@@ -0,0 +1,204 @@
+<!--
+//  LaunchOptions.vue
+//
+//  Created by KasenVR on 19 May 2020.
+//  Copyright 2020 Vircadia and contributors.
```
Comment (ctrlaltdavid) [inline src/components/Dialogs/LaunchOptions.vue]:
```suggestion
//  Copyright 2020 Vircadia contributors.
```

File: src/components/Dialogs/LaunchOptions.vue [hunk 48]
```diff
@@ -0,0 +1,204 @@
...
+            set(value) {
+                this.$store.commit('mutate', {
+                    property: 'dontPromptForLogin', 
+                    with: value
+                });
+            },
+        },
+    },
+    created: function () {
+
```
Comment (ctrlaltdavid) [inline src/components/Dialogs/LaunchOptions.vue]:
```suggestion
        // Nothing to do.
```

File: src/components/Dialogs/NoInstallerFound.vue [hunk 49]
```diff
@@ -24,16 +24,15 @@
             </v-card-title>
     
             <v-card-text>
-                An installer for interface was not found. Try downloading it by pressing the "Download Interface" button.
+                An installer for Interface was not found.
             </v-card-text>
     
             <v-divider></v-divider>
-    
+
```
Comment (ctrlaltdavid) [inline src/components/Dialogs/NoInstallerFound.vue]:
```suggestion

```
And similar, below.

File: src/components/Dialogs/ReportAnIssue.vue [hunk 50]
```diff
@@ -0,0 +1,76 @@
+<!--
+//  ReportAnIssue.vue
+//
+//  Created by KasenVR on 19 May 2020.
+//  Copyright 2020 Vircadia and contributors.
```
Comment (ctrlaltdavid) [inline src/components/Dialogs/ReportAnIssue.vue]:
```suggestion
//  Copyright 2020 Vircadia contributors.
```

File: src/components/Dialogs/ReportAnIssue.vue [hunk 51]
```diff
@@ -0,0 +1,76 @@
...
+
+<script>
+const { ipcRenderer } = require('electron');
+
+export default {
+    name: 'ReportAnIssue',
+	methods: {
+        reportAnIssue: function (url) {
+            const { shell } = require('electron');
+			shell.openExternal(url);
```
Comment (ctrlaltdavid) [inline src/components/Dialogs/ReportAnIssue.vue]: Tabs.

File: src/components/News.vue [hunk 52]
```diff
@@ -0,0 +1,46 @@
+<!--
+//  News.vue
+//
+//  Created by Kalila L. on 11 May 2020.
+//  Copyright 2020 Vircadia and contributors.
```
Comment (ctrlaltdavid) [inline src/components/News.vue]:
```suggestion
//  Copyright 2020 Vircadia contributors.
```

File: src/components/News.vue [hunk 53]
```diff
@@ -0,0 +1,46 @@
...
+export default {	
+	name: 'News',
+	methods: {
+	},
+	data: () => ({
+	}),
+    computed: {
+    },
+    watch: {
+    },
```
Comment (ctrlaltdavid) [inline src/components/News.vue]:
```suggestion
    methods: { },
    data: () => ({ }),
    computed: { },
    watch: { },
```

File: src/components/News.vue [hunk 54]
```diff
@@ -0,0 +1,46 @@
...
+	created: function () {
```
Comment (ctrlaltdavid) [inline src/components/News.vue]: Tabs.

File: src/main.js [hunk 55]
```diff
@@ -8,6 +8,9 @@
     See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
 */
 
+// import * as Sentry from '@sentry/electron';
+// Sentry.init({dsn: 'https://def94db0cce14e2180e054407e551220@sentry.vircadia.dev/3'});
```
Comment (ctrlaltdavid) [inline src/main.js:12]: Remove commented-out code.

File: src/components/Settings.vue [hunk 56]
```diff
@@ -97,39 +111,40 @@
...
                         
                         <v-divider
                             class="mx-3"
                         ></v-divider>
 						
-                        <h3 class="mx-7 mt-5">Settings for {{ selectedInterface }}</h3>
+                        <h3 v-show="false" class="mx-7 mt-5">Settings for {{ selectedInterface }}</h3>
                         
-						<v-layout row pr-5 pt-5 pl-12>
+						<v-layout v-show="false" row pr-5 pt-5 pl-12>
```
Comment (ctrlaltdavid) [inline src/components/Settings.vue]: Tabs.

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

File: src/main.js (hunk 55 above)
Comment (two-one-five) [reply src/main.js:12]: For now, I keep it there to uncomment in the parts that it is for testing builds distributed to non-technical users for feedback.

File: src/background.js (hunk 27 above)
Comment (two-one-five) [reply src/background.js:185]: This code is here for if/when different methods of finding information is needed.

File: src/background.js (hunk 30 above)
Comment (two-one-five) [reply src/background.js:279]: This will be used again in the future most likely.

File: public/css/styles.css (hunk 14 above)
Comment (two-one-five) [reply public/css/styles.css:65]: Yep!

**Review by ctrlaltdavid — APPROVED** [review]

**Pull request merged**